[RFC] Force je_ prefix for jemalloc functions, small CMake fixes.

fwalch opened the pull request:
**CMake**
- Refer to Unibilium instead of LibUnibilium.
- Remove shared libraries to link dependencies statically.
- Remove duplicate PkgConfig lookups.

**jemalloc**
- Set JEMALLOC_NO_DEMANGLE to be able to use `je_*` functions,
  regardless of how jemalloc was compiled (--with-jemalloc-prefix)
- Show jemalloc information in Neovim's version output.

Resolve #2449.

fwalch committed:
CMake: Refer to Unibilium instead of LibUnibilium.

fwalch committed:
jemalloc: Force use of prefixed functions.

* Set JEMALLOC_NO_DEMANGLE to be able to use `je_*` functions,
  regardless of how jemalloc was compiled (--with-jemalloc-prefix)
* Show jemalloc information in Neovim's version output.

Resolve #2449.

fwalch committed:
CMake: Remove duplicate PkgConfig lookups.

fwalch committed:
CMake: Don't request static linking for dependencies.

For built-in dependencies, shared libraries are removed and static
linking is always used.

For systemwide dependencies, static linking should not be used.

jszakmeister commented on CMakeLists.txt:
```diff
@@ -178,50 +178,49 @@ endif()
 
+# Systemwide (non-bundled) dependencies are usually not provided
+# as static libraries, so we can just switch this on by default.
+option(USE_STATIC "Use static dependencies" ON)
```
I was trying hard to avoid doing this for two reasons.  First, now it's another knob distros need to be aware of.  Second, because `OFF` doesn't make sense for when we use bundled dependencies (we don't install the dependent libraries anywhere useful).  By only having static libraries available from the third party build, we'd get the behavior we wanted, and things fell back to shared libraries without explicitly asking for it.  That was the hope anyways.

jszakmeister commented:
The rest of this looks good to me.  Thanks for digging into jemalloc and finding that you could control the mangling differently.

fwalch committed:
deps: Remove shared libraries.

fwalch commented:
@jszakmeister Updated, let me know what you think :-)

tarruda commented:
LGTM

Pyrohh commented:
Totally up to you @fwalch, but the flags applied by `-DSANITIZE=ON` don't show up in `nvim --version`; perhaps that could be fixed here? I'm only mentioning so because you seem to be doing the same with jemalloc, but I'm not sure if it's too much work or if you want to.

fwalch commented:
@Pyrohh I'm not sure if it's possible, because these flags only get applied to a certain target, not the whole build.

Pyrohh commented:
Before you merge, make sure to remove the commit regarding the Travis deps.

fwalch commented:
First got to make sure the builds actually pass :-D thanks for the heads up!

fwalch commented:
```
/usr/bin/gcc -m32 -shared -o xml/core.so -L/home/travis/build/neovim/neovim/.deps/usr/lib src/Parser.o src/bind/dub/dub.o src/bind/xml_Parser.o src/bind/xml_core.o -lstdc++

/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/4.6/libstdc++.so when searching for -lstdc++

/usr/bin/ld: skipping incompatible /usr/lib/gcc/x86_64-linux-gnu/4.6/libstdc++.a when searching for -lstdc++

/usr/bin/ld: cannot find -lstdc++

collect2: ld returned 1 exit status
```

Any ideas what caused this?

Pyrohh commented:
Shooting from the hip here, but maybe related to #2431?

fwalch commented:
Thought so as well, but bot-ci deps32 builds work..

fwalch commented:
Okay, so the 32-bit build has been fixed by #2515. But I can't get the GCC build to work, it always times out...

fwalch commented:
The other builds timed out with "Received no output for 10 minutes", but the latest one got "Your test run exceeded 50 minutes."

The GCC build on master takes 42 minutes (!!), which is already quite close to the 50min max. Building the deps as well seems to be too much.

justinmk commented:
restarted :/

fwalch commented:
Exceeded the build time again.. we need to figure something out. Maybe #2340 will help, the GCC build there "only" takes 30min (including building the dependencies).

How should I proceed with this PR? Just remove the "build deps" test commit and merge?

Tranquility commented:
@fwalch thanks for the work. It would be nice to see this merged.

fwalch commented:
Rebased. The [build](https://travis-ci.org/neovim/neovim/builds/60588543) didn't change (passed except timeout for gcc job). Removed the temporary commit to build the deps on Travis.

@justinmk Do you think this can be merged now? (the build will fail because it uses `neovim/deps`).
